lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -181,168 +186,484 @@ def test_duplicate(self, selinux_disabled, server_config, make_logger, tarball):
-        assert results_link.is_symlink()
-        assert results_link.samefile(incoming_dir)
-
-        assert tree.datasets[md5].unpacked == incoming_dir
-        assert tree.datasets[md5].results_link == results_link
-
-        # Re-discover, with all the files in place, and compare
-        newtree = FileTree(server_config, make_logger)
-        newtree.full_discovery()
-
-        assert newtree.archive_root == tree.archive_root
-        assert newtree.incoming_root == tree.incoming_root
-        assert newtree.results_root == tree.results_root
-        assert sorted(newtree.controllers) == sorted(tree.controllers)
-        assert sorted(newtree.datasets) == sorted(tree.datasets)
-        assert sorted(newtree.tarballs) == sorted(tree.tarballs)
-        for controller in tree.controllers.values():
-            other = newtree.controllers[controller.name]
-            assert controller.name == other.name
-            assert controller.path == other.path
-            assert sorted(controller.datasets) == sorted(other.datasets)
-            assert sorted(controller.tarballs) == sorted(other.tarballs)
-        for tarball in tree.datasets.values():
-            other = newtree.datasets[tarball.resource_id]
-            assert tarball.name == other.name
-            assert tarball.resource_id == other.resource_id
-            assert tarball.controller_name == other.controller_name
-            assert tarball.tarball_path == other.tarball_path
-            assert tarball.md5_path == other.md5_path
-            assert tarball.unpacked == other.unpacked
-            assert tarball.results_link == other.results_link
+            # Expect the archive directory was created, but we haven't unpacked so
+            # incoming and results should not exist.
+            assert archive.is_dir()
+            assert not incoming.exists()
+            assert not results.exists()
+
+            # The original files should have been removed
+            assert not source_tarball.exists()
+            assert not source_md5.exists()
+
+            tarfile = archive / source_tarball.name
+            md5file = archive / source_md5.name
+            assert tarfile.exists()
+            assert md5file.exists()
+
+            todo_state = archive / "TODO" / tarfile.name
+            assert todo_state.is_symlink()
+            assert todo_state.samefile(tarfile)
+
+            assert md5 == md5file.read_text()
+            hash = hashlib.md5()
+            hash.update(tarfile.read_bytes())
+            assert md5 == hash.hexdigest()
+
+            assert list(tree.controllers.keys()) == ["ABC"]
+            dataset_name = source_tarball.name[:-7]
+            assert list(tree.tarballs) == [dataset_name]
+            assert list(tree.datasets) == [md5]
+
+            # Now "unpack" the tarball and check that the incoming directory and
+            # results link are set up.
+            incoming_dir = incoming / dataset_name
+            incoming_dir.mkdir(parents=True)
```

Comment:
It seems odd that this change is required...is it actually required?  If so, why?

(It is generally a Bad Thing™ to change _both_ the CUT and the unit test code at the same time.  Obviously, for new code we need new unit tests, but for existing tests of existing code, we're supposed to be able to run the tests _unchanged_ on the new code and have them pass, unless the "interface contract" has changed, but I'm not aware of that being the case, here....)

Reply:
Corrected it..